provision/initramfs/capabilities/setup-filesystems/20-filesystems

```diff
@@ -108,6 +110,65 @@ run_mkfs() {
   esac
 }
 
+run_mdadm() {
+  local LEVEL="${1}"
+  local NUMBER="${2}"
+  
+  if [ -z "${LEVEL}" ]; then
+    wwlogger "No raid level specified"
+    exit 2
+  fi
+  
+  if [ -z "${NUMBER}" ]; then
+    wwlogger "No raid number given for raid${LEVEL}"
+    exit 2
+  fi
+  
+  shift 2
+  local COUNT="$#"
+  local DEVICES
+  local DEV
+
+  if [ ${COUNT} -eq 0 ]; then
+    wwlogger "No devices specified for raid${LEVEL} on /dev/md${NUMBER}"
+    exit 2
+  fi
+
+  msg_gray "   * creating raid${LEVEL} on /dev/md${NUMBER} for ${*}"
+
+  for DEV in ${@:-}; do
+    case "${DEV}" in
+      missing|-)
+        DEVICES="${DEVICES} missing"
+      ;;
+      /dev/*)
+        if [ -x /usr/bin/udevadm ]; then
+          /usr/bin/udevadm settle --timeout=15 --exit-if-exists="${DEV}"
+        fi
+        DEVICES="$DEVICES ${DEV}"
+      ;;
+      *)
+        if [ -x /usr/bin/udevadm ]; then
+          /usr/bin/udevadm settle --timeout=15 --exit-if-exists="/dev/${DEV}"
+        fi
+        DEVICES="$DEVICES /dev/${DEV}"
+      ;;
+    esac
+  done
+
+  if [ ! -x /usr/bin/udevadm ]; then
+    sleep 1
+  fi
+
+  {
+    # This is ugly but it seems necessary
+    (/usr/bin/yes || /bin/true) | /sbin/mdadm --create "/dev/md${NUMBER}" -f "--level=${LEVEL}" --raid-devices=${COUNT} ${DEVICES} >/dev/null 2>&1 && wwsuccess
```

Comment:
Assuming that `--force` isn't sufficient to get mdadm to not prompt?

Double quote ${DEVICES} to avoid word splitting.

Reply:
"--force" isn't sufficient. I don't like it, but `yes` seems the only way...

Reply:
And I don't think double-quoting `${DEVICES}` is right. We want it bo be multiple parameters, not just one with spaces in it.

Reply:
> "--force" isn't sufficient. I don't like it, but `yes` seems the only way...

Ok. Please update the comment above stating what this is a workaround for.

> And I don't think double-quoting `${DEVICES}` is right. We want it bo be multiple parameters, not just one with spaces in it.

Yep you're correct.